runtime/master/src/main/java/org/apache/nemo/runtime/master/resource/ContainerManager.java

```diff
@@ -149,18 +149,39 @@ public void onContainerAllocated(final String executorId,
     }
 
     final ResourceSpecification resourceSpecification = selectResourceSpecForContainer();
+    final List<Configuration> configurationsToMerge = new ArrayList<>();
+
     evaluatorIdToResourceSpec.put(allocatedContainer.getId(), resourceSpecification);
 
     LOG.info("Container type (" + resourceSpecification.getContainerType()
       + ") allocated, will be used for [" + executorId + "]");
     pendingContextIdToResourceSpec.put(executorId, resourceSpecification);
 
+    configurationsToMerge.add(executorConfiguration);
+
+    // ExecutorMemory handling
+    configurationsToMerge.add(Tang.Factory.getTang().newConfigurationBuilder()
+        .bindNamedParameter(JobConf.ExecutorMemoryMb.class, String.valueOf(resourceSpecification.getMemory()))
+        .build()
+    );
+
+    // MaxOffheapRatio handling
+    resourceSpecification.getMaxOffheapRatio().ifPresent(value -> {
+      configurationsToMerge.add(Tang.Factory.getTang().newConfigurationBuilder()
+        .bindNamedParameter(JobConf.MaxOffheapRatio.class, String.valueOf(value))
+        .build()
+      );
+    });
+
     // Poison handling
-    final Configuration poisonConfiguration = Tang.Factory.getTang().newConfigurationBuilder()
-      .bindNamedParameter(JobConf.ExecutorPosionSec.class, String.valueOf(resourceSpecification.getPoisonSec()))
-      .build();
+    resourceSpecification.getPoisonSec().ifPresent(value -> {
```

Comment:
Let's get rid of the `{` and `}`, and `;`, as it causes a sonar cloud issue.

Reply:
good point, thank you.